### PR TITLE
Update optimizations.md

### PR DESCRIPTION
**Pull request by teesloane**
Updates to the optimizations doc:

- add bigger intro/outro
- Add workflows, workspaces, resource_class

### CODE REVIEW

**Comment by teesloane** [comment]
This is ready for a review now @michelle-luna.

File: jekyll/_cci2/optimizations.md [hunk 1]
```diff
@@ -6,33 +6,108 @@ description: "CircleCI 2.0 build optimizations"
...
+Workflows provide a means to define a collection of jobs and their run order. If
+at any point in your build you see a step where two jobs could happily run
+independent of one another, workflows may be helpful. Further, workflows
+allow you to schedule jobs, run jobs in parallel, or simply allow you runs jobs
+independently with easier troubleshooting.
+
+**Note**: Workflows are available to all plans, but running parallel jobs assumes that
+your plan provides multiple machines to execute on.
+
+```yaml
```
Comment (michelle-luna) [inline jekyll/_cci2/optimizations.md]: can we add some comments to this example?

File: jekyll/_cci2/optimizations.md [hunk 2]
```diff
@@ -6,33 +6,108 @@ description: "CircleCI 2.0 build optimizations"
...
+
+Workspaces are used to pass along data that is _unique to a run_ and is needed for _downstream jobs_. So, if you are using workflows, a job run earlier in your build might fetch data and then make it _available later_ for jobs that run later in a build.
+
+To persist data from a job and make it available to other jobs, configure the job to use the [`persist_to_workspace`]({{ site.baseurl}}/2.0/configuration-reference#persist_to_workspace) key. Files and directories named in the paths: property of `persist_to_workspace` will be uploaded to the workflow’s temporary workspace relative to the directory specified with the root key. The files and directories are then uploaded and made available for subsequent jobs (and re-runs of the workflow) to use.
+
+Read more about how to use workspaces in the [workflows document]({{site.baseurl}}/2.0/workflows/#using-workspaces-to-share-data-among-jobs).
+
+## Parallelism
+
+**Note**: This feature is only available on paid plans.
```
Comment (michelle-luna) [inline jekyll/_cci2/optimizations.md]: Is this true? I don't know why this seems untrue to me...

File: jekyll/_cci2/optimizations.md [hunk 3]
```diff
@@ -42,26 +117,66 @@ jobs:
...
       - run: docker build .
-```      
+```
+
+Learn more about [Docker Layer Caching]({{site.baseurl}}/2.0/docker-layer-caching)
 
 ## See Also
+{:.no_toc}
+
+If you still have questions optimizing your build configuration consider consulting our [FAQ]({{ site.baseurl }}/2.0/faq) or [opening a support ticket](https://support.circleci.com/hc/en-us/requests/new).
```
Comment (michelle-luna) [inline jekyll/_cci2/optimizations.md]: Move this support link up to where we mention it for DLC

**Pull request merged**